infrastructure/src/main/java/org/corfudb/infrastructure/HandlerMethods.java

```diff
@@ -151,17 +128,17 @@ private void registerMethod(@Nonnull final MethodHandles.Lookup caller,
                 + method.getParameterTypes()[0]);
         }
         if (handlerMap.containsKey(annotation.type())) {
-            throw new UnrecoverableCorfuError("Handler for " + annotation.type()
+            throw new UnrecoverableCorfuError("HandlerMethod for " + annotation.type()
                 + " already registered!");
         }
         // convert the method into a Java8 Lambda for maximum execution speed...
         try {
-            Handler<CorfuMsg> h;
+            HandlerMethod<CorfuMsg> h;
             if (Modifier.isStatic(method.getModifiers())) {
                 MethodHandle mh = caller.unreflect(method);
                 MethodType mt = mh.type().changeParameterType(0, CorfuMsg.class);
-                h = (Handler<CorfuMsg>) LambdaMetafactory.metafactory(caller,
-                    "handle", MethodType.methodType(Handler.class),
+                h = (HandlerMethod<CorfuMsg>) LambdaMetafactory.metafactory(caller,
```

Comment:
as per our discussion, at some point we should get rid of these meta introspections so that we do not pay cost in fast paths

Reply:
I already have a patch to remove reflection from RPC, but I need to push in my thread management PRs first.